rapid7/metasploit-framework
# Add a post-module and a mixin for communicating with F5's MCP

Check in a mixin that implements F5's "MCP" database protocol, and three POST-modules that pull information from them.

Everything is tested against F5 Big-IP 17.0.0.1, which is freely available as a VM image (with a free account). It's also in our "vulnerable software" directory.

## Verification

1. Install the application
2. Start `msfconsole`
3. Do: Get any session somehow (`exploit/linux/http/f5_icontrol_rpmspec_rce_cve_2022_41800` from the other branch works well on 17.0.0.1 and earlier, or just use `msfvenom`)
4. Do: `use` any of the three post-modules (`use post/linux/gather/f5_enum_users` / `use post/linux/gather/f5_enum_db_variables` / `use post/linux/gather/f5_password`)
5. Do `set SESSION <sessionid>`
6. Do: `run`
7. You should get the info
